### ISSUE TITLE
Do we supprot polynomial regression put decimals predict variable in pow?

### ISSUE THREAD

**Issue by Iteravse** [tensorflow/tfjs#6933]
i discover the problem really is ,but not answer yet,
when i put the decimals predict varible in pow function,the predict result will always return NAN,here's codepen [example](https://codepen.io/iteravse/pen/eYVKyyR):
and if that runing at codepen,it returns null,need waiting some time
img
any idea? it work only when i remove decimals predict variable in pow,but i need that

acitally this is a old quesition,in [here](https://github.com/tensorflow/tfjs/issues/6474)

**Comment by pyu10055** [comment]
@Iteravse Thank you for reporting this, I tried your example, the reason of `NaN` is likely caused by the learning rate is too high that cause the power base coefficient `a` to turn into a negative number. As that happens, the value of `pow(a, b)`, if `b` is not a integer, will be a complex number. This is reason for the coefficient to become `NaN`.

If you lower the learning rate to 0.003, you will see the optimizer will not produce `NaN` values, but it does not mean the training will converge.

**Comment by Iteravse** [comment]
> 

thanks for a lot,but how to ensure it converge?

**Comment by pyu10055** [comment]
> > 
> 
> @pyu10055 thanks for a lot,but how to ensure it converge?

I am not a modeling expert, I suspect you need to convert the formula to a polynomial term (i.e. take a log of the whole thing), to ensure a global minimum for the loss function.

**Comment by Iteravse** [comment]
> 

you mean we need discard pow function in predict formula? replace it by none pow func polynomial?

**Comment by gaikwadrahul8** [comment]
Hi, @Iteravse 

Apologize for the delayed response and could you please refer above comment from @pyu10055 and May I know whether above comment resolving your issue or not ? If issue still persists please let us know ? Thank you!
